network_diffusion/models/ICM_model.py

```diff
@@ -0,0 +1,223 @@
+
+from typing import Dict, List, Tuple
+import random
+import numpy as np
+import networkx as nx
+from network_diffusion.mln.mlnetwork import MultilayerNetwork
+from network_diffusion.models.base_model import BaseModel
+from network_diffusion.mln.actor import MLNetworkActor
+from network_diffusion.models.utils.compartmental import CompartmentalGraph
+from network_diffusion.models.utils.types import NetworkUpdateBuffer as NUBff
+from network_diffusion.seeding.base_selector import BaseSeedSelector
+from network_diffusion.utils import BOLD_UNDERLINE, THIN_UNDERLINE, NumericType
+
+
+class ICModel(BaseModel):
```

Comment:
I'd leave this class as a baseline implementation and try to focus on your second (much mode clear) approach :)